docs/interfaces/auth-service.txt

```diff
@@ -0,0 +1,99 @@
+Interface Token {
+    payload: {
+        // The lifetime of the Token in milliseconds. The date in unix seconds of the expiration.
+        expiresIn: number (ms),
+        // The corresponding userId of the requesting client.
+        userId: number,
+        // The id of the current session.
+        sessionId: string
+    },
+    signature: string
+}
+
+Interface Cookie {
+    // The id for the session corresponding to the client, who has signed in.
+    sessionId: string;
+}
+
+/**
+* The client has not the necessary permissions for the requesting action.
+*/
+Exception NoPermissions
+
+/**
+* The credentials for login/authentication are not valid.
+*/
+Exception InvalidCredentials
+
+/**
+* The client can login with its credentials for authentication.
+* If they are correct, the service answers with a signed Token and sets a cookie, containing the sessionId of the client.
+*
+* If they aren't correct, the service throws an error.
+*
+* @throws InvalidCredentials
+*/
+Login (username: string, password: string): (Token, Cookie);
+
+/**
+* An example for any protected route. If the client requests protected resources, it has to 
+* send the signed Token and the cookie, it receives from the service at login, to the server.
+*
+* This will be a library to act as part of the auth-service. The other services have not to 
+* request the auth-service for authentication. Instead, they use this library-function in their own
+* and receive knowledge about authentication without request.
+*
+* Throws an exception, if the token is not valid. E.g. if the signature is not valid.
+*
+* @throws InvalidCredentials
+*/
+Authenticate (token: Token, cookie: Cookie): {userId: number; sessionId: string;};
+
+/**
+* A request to get knowledge about themselves. This information is contained in the payload of 
+* a Token. So, this function handles the refreshing of a Token.
+*
+* Sends back a new Token.
+*
+* Throws an exception, if the cookie is empty or the transmitted sessionId is wrong.
+*
+* @throws InvalidCredentials
+*/
+WhoAmI(cookie: Cookie): Token;
+
+/**
+* Function to kill one specific session by its id.
```

Comment:
Just a thought: You must check the users permissions, if the given sessionId does not belong to the user in the token. So a user can always end his own sesions, but osers cannot end other users sessions. But there is an exception: If the user (given in the token) is a superadmin, he can clear sessions of other users.

This means, that the auth service will need the permission service at this point (which is not tragic, but just for considerations).